### ISSUE TITLE
Markdown-examples don't build properly

### ISSUE THREAD

**Issue by FMNSSun** [FiwareULPGC/fiware-markdown-to-pdf#2]
The conversion from example markdown files fails:

```
$ md2pdf -i mkdocs_user-cover.yml -o out.pdf
Cover metadata not provided. Trying to generate it
Processing file [user-cover/index.md] ...
Processing file [user-cover/index.md] ...OK
Processing file [user-cover/README.md] ...
Processing file [user-cover/README.md] ...OK
Processing file [user-cover/pagination.md] ...
Processing file [user-cover/pagination.md] ...OK
Processing file [user-cover/geolocation.md] ...
Processing file [user-cover/geolocation.md] ...OK
Processing file [user-cover/structured_attribute_valued.md] ...
Processing file [user-cover/structured_attribute_valued.md] ...OK
Processing file [user-cover/context_providers.md] ...
Processing file [user-cover/context_providers.md] ...OK
Processing file [user-cover/metadata.md] ...
Processing file [user-cover/metadata.md] ...OK
Processing file [user-cover/filtering.md] ...
Processing file [user-cover/filtering.md] ...OK
Processing file [user-cover/multitenancy.md] ...
Processing file [user-cover/multitenancy.md] ...OK
Processing file [user-cover/service_path.md] ...
Processing file [user-cover/service_path.md] ...OK
Processing file [user-cover/append_and_delete.md] ...
Processing file [user-cover/append_and_delete.md] ...OK
Processing file [user-cover/updating_regs_and_subs.md] ...
Processing file [user-cover/updating_regs_and_subs.md] ...OK
Processing file [user-cover/federation.md] ...
Processing file [user-cover/federation.md] ...OK
Processing file [user-cover/duration.md] ...
Processing file [user-cover/duration.md] ...OK
Processing file [user-cover/empty_types.md] ...
Processing file [user-cover/empty_types.md] ...OK
Processing file [user-cover/std_conv_mix.md] ...
Processing file [user-cover/std_conv_mix.md] ...OK
Processing file [user-cover/http_and_ngsi_sc.md] ...
Processing file [user-cover/http_and_ngsi_sc.md] ...OK
Processing file [user-cover/security.md] ...
Processing file [user-cover/security.md] ...OK
Processing file [user-cover/known_limitations.md] ...
Processing file [user-cover/known_limitations.md] ...OK
Processing file [user-cover/forbidden_characters.md] ...
Processing file [user-cover/forbidden_characters.md] ...OK
Processing file [user-cover/walkthrough_apiv1.md] ...
Processing file [user-cover/walkthrough_apiv1.md] ...OK
Processing file [user-cover/v1_v2_coexistence.md] ...
Processing file [user-cover/v1_v2_coexistence.md] ...OK
Generating PDF...
Traceback (most recent call last):
  File "/usr/local/bin/md2pdf_pandoc_filter", line 9, in <module>
    load_entry_point('markdown-to-pdf==0.1', 'console_scripts', 'md2pdf_pandoc_filter')()
  File "/usr/local/lib/python2.7/dist-packages/markdown_to_pdf-0.1-py2.7.egg/markdown_to_pdf/pandoc_filters.py", line 122, in main
    toJSONFilter(pandoc_filter)
  File "/usr/local/lib/python2.7/dist-packages/pandocfilters.py", line 46, in toJSONFilter
    toJSONFilters([action])
  File "/usr/local/lib/python2.7/dist-packages/pandocfilters.py", line 76, in toJSONFilters
    altered = reduce(lambda x, action: walk(x, action, format, doc[0]['unMeta']), actions, doc)
  File "/usr/local/lib/python2.7/dist-packages/pandocfilters.py", line 76, in <lambda>
    altered = reduce(lambda x, action: walk(x, action, format, doc[0]['unMeta']), actions, doc)
  File "/usr/local/lib/python2.7/dist-packages/pandocfilters.py", line 34, in walk
    array.append(walk(item, action, format, meta))
  File "/usr/local/lib/python2.7/dist-packages/pandocfilters.py", line 27, in walk
    array.append(walk(item, action, format, meta))
  File "/usr/local/lib/python2.7/dist-packages/pandocfilters.py", line 39, in walk
    obj[k] = walk(x[k], action, format, meta)
  File "/usr/local/lib/python2.7/dist-packages/pandocfilters.py", line 25, in walk
    res = action(item['t'], item['c'], format, meta)
  File "/usr/local/lib/python2.7/dist-packages/markdown_to_pdf-0.1-py2.7.egg/markdown_to_pdf/pandoc_filters.py", line 105, in pandoc_filter
    return Image(value[0],value[1])
  File "/usr/local/lib/python2.7/dist-packages/pandocfilters.py", line 119, in fun
    + str(lenargs))
ValueError: Image expects 3 arguments, but given 2
pandoc: Error running filter md2pdf_pandoc_filter
Filter returned error status 1
ERROR:  Conversion to PDF failed - Pandoc failed with code: (83)

```

as for the versions of the dependencies used:

```
Processing dependencies for markdown-to-pdf==0.1
Searching for pandocfilters==1.3.0
Best match: pandocfilters 1.3.0
pandocfilters 1.3.0 is already the active version in easy-install.pth

Using /usr/local/lib/python2.7/dist-packages
Searching for pypandoc==1.1.2
Best match: pypandoc 1.1.2
pypandoc 1.1.2 is already the active version in easy-install.pth

Using /usr/local/lib/python2.7/dist-packages
Searching for Markdown==2.6.5
Best match: Markdown 2.6.5
Markdown 2.6.5 is already the active version in easy-install.pth

Using /usr/local/lib/python2.7/dist-packages
Searching for PyYAML==3.11
Best match: PyYAML 3.11
PyYAML 3.11 is already the active version in easy-install.pth
```

The ValueError is raised by the `elt` function from pandocfilters when used in `return Image(value[0],value[1])`. 

The same kind of ValueError can be triggered for different things as well when using different markdown documents. For example the `return Link(value[0],value[1])` (pandoc_filters.py) will trigger the same Error. 

This is due to that `pandocfilters.py` defines both Image and Link to have three arguments

```
Link = elt('Link', 3)
Image = elt('Image', 3)
```

but `pandoc_filters.py` only passes two arguments to these. 

Additional version info:

```
$ python --version
Python 2.7.3
$ pandoc --version
pandoc 1.15.1
Compiled with texmath 0.8.4, highlighting-kate 0.6.
$ pdftk --version

pdftk 2.02 a Handy Tool for Manipulating PDF Documents
```

**Comment by FiwareULPGC** [comment]
Hi!

first of all, thanks for your detailed report. 

The cause of the error is that Pandoc has released this January its version 1.16 and it breaks backward compatibility with images and links filters. "pandocfilters" library has been updated to maintain compability with Pandoc but it also breaks backward compability.

In order to solve this issue, you should uninstall your current version of pandocfilters and install the 1.2.4 version instead. You can do that using pip as follows:

```
sudo pip uninstall pandocfilters
sudo pip install "pandocfilters<=1.2.4"
```

We will also release a new version with the correct dependencies today.

Regards.

**Comment by FMNSSun** [comment]
Hi, thanks for the fast reply. Your suggestions resolve the issue and PDF generation works fine now.

**Comment by FiwareULPGC** [comment]
Solved in 15a504f213befdfa0650caf3307f66e1d6e079d3
